### PR TITLE
Fix Nothing DataType serde

### LINKED ISSUE #389: "select null as aa,null as bb" is not supported in ClickHouse-Native-JDBC
### Environment

- OS version: SUSE 12.5
- JDK version:1.8
- ClickHouse Server version: 20.3.2.1
- ClickHouse Native JDBC version:2.5.4

### Error logs
Exception in thread "main" java.lang.UnsupportedOperationException: Unknown compression method: 108
	at com.github.housepower.jdbc.buffer.CompressedBuffedReader.readCompressedData(CompressedBuffedReader.java:94)
	at com.github.housepower.jdbc.buffer.CompressedBuffedReader.readBinary(CompressedBuffedReader.java:50)
	at com.github.housepower.jdbc.serde.BinaryDeserializer.readUTF8StringBinary(BinaryDeserializer.java:90)
	at com.github.housepower.jdbc.data.Block.readFrom(Block.java:40)
	at com.github.housepower.jdbc.protocol.DataResponse.readFrom(DataResponse.java:32)
	at com.github.housepower.jdbc.protocol.Response.readFrom(Response.java:33)
	at com.github.housepower.jdbc.connect.NativeClient.receiveResponse(NativeClient.java:178)
	at com.github.housepower.jdbc.connect.NativeClient.lambda$receiveQuery$0(NativeClient.java:135)
	at com.github.housepower.jdbc.stream.ClickHouseQueryResult.consumeDataResponse(ClickHouseQueryResult.java:89)
	at com.github.housepower.jdbc.stream.ClickHouseQueryResult.access$100(ClickHouseQueryResult.java:26)
	at com.github.housepower.jdbc.stream.ClickHouseQueryResult$1.fill(ClickHouseQueryResult.java:65)
	at com.github.housepower.jdbc.stream.ClickHouseQueryResult$1.hasNext(ClickHouseQueryResult.java:55)
	at com.github.housepower.jdbc.ClickHouseResultSet.fetchBlock(ClickHouseResultSet.java:439)
	at com.github.housepower.jdbc.ClickHouseResultSet.next(ClickHouseResultSet.java:408)

### Steps to reproduce
Use the ClickHouse-Native-JDBC (2.5.4) driver to access the ClickHouse database, construct the query "select null as aa,null as bb", and use the ResultSet to store the query result. After the query is executed, the preceding error is reported.

### Other descriptions
At first, I used ClickHouse-Native-JDBC 2.2. When I used the “select null as aa” syntax, an error message was displayed, and the error message was " Exception in thread "main" java.sql.SQLException: Unknown data type family:Nothing". Later, when I upgraded to 2.5.4, If one "null as xx" is displayed in the query result, no problem occurs. If multiple "null as xx" are displayed, an error message is displayed, as shown in the Error logs.

### PR DESCRIPTION
**Pull request by sundy-li**
Fixes #389

### CODE REVIEW

File: clickhouse-native-jdbc/src/main/java/com/github/housepower/data/type/complex/DataTypeNothing.java [hunk 1]
```diff
@@ -14,20 +14,21 @@
...
 import com.github.housepower.client.NativeContext;
 import com.github.housepower.data.IDataType;
+import com.github.housepower.exception.InvalidOperationException;
 import com.github.housepower.misc.SQLLexer;
 import com.github.housepower.serde.BinaryDeserializer;
 import com.github.housepower.serde.BinarySerializer;
 
+import java.io.IOException;
+import java.sql.SQLException;
+import java.sql.Types;
```
Comment (pan3793) [inline clickhouse-native-jdbc/src/main/java/com/github/housepower/data/type/complex/DataTypeNothing.java:26]: unnecessary change

File: clickhouse-native-jdbc/src/main/java/com/github/housepower/data/type/complex/DataTypeNothing.java [hunk 2]
```diff
@@ -14,20 +14,21 @@
...
+
 public class DataTypeNothing implements IDataType<Object, Object> {
 
     public static DataTypeCreator<Object, Object> CREATOR =
-            (lexer, serverContext) -> new DataTypeNothing(serverContext);
+        (lexer, serverContext) -> new DataTypeNothing(serverContext);
```
Comment (pan3793) [inline clickhouse-native-jdbc/src/main/java/com/github/housepower/data/type/complex/DataTypeNothing.java:31]: unnecessary change

File: clickhouse-native-jdbc/src/main/java/com/github/housepower/data/type/complex/DataTypeNothing.java [hunk 3]
```diff
@@ -63,12 +64,16 @@ public int getScale() {
     }
 
     @Override
-    public void serializeBinary(Object data, BinarySerializer serializer) throws SQLException, IOException {
+    public void serializeBinary(Object data, BinarySerializer serializer)
+        throws SQLException, IOException {
```
Comment (pan3793) [inline clickhouse-native-jdbc/src/main/java/com/github/housepower/data/type/complex/DataTypeNothing.java:68]: unnecessary change

**Comment by pan3793** [comment]
Let's avoid unnecessary changes to make this patch portable

**Review by pan3793 — APPROVED** [review]

File: clickhouse-native-jdbc/src/main/java/com/github/housepower/data/type/complex/DataTypeNothing.java (hunk 1 above)
Comment (pan3793) [reply clickhouse-native-jdbc/src/main/java/com/github/housepower/data/type/complex/DataTypeNothing.java:26]: the sort order changed again...

**Comment by wallace-clickhouse** [comment]
@sundy-li @pan3793 
Could the community launch a release package based on the fixed version?
Thank you!

**Pull request merged**